rclpy/test/rostopic_list.py

```diff
@@ -0,0 +1,34 @@
+# Copyright 2016 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import time
+
+import rclpy
+from rclpy.impl.rmw_implementation_tools import select_rmw_implementation
+from rclpy.impl.rmw_implementation_tools import get_rmw_implementations
+
+
+rmw_implementations = get_rmw_implementations()
+if os.environ.get('RCLPY_IMPLEMENTATION', 'Not Set') in rmw_implementations:
+    select_rmw_implementation(os.environ['RCLPY_IMPLEMENTATION'])
```

Comment:
is this necessary? seems to be taken care of by [`rclpy.init`](https://github.com/ros2/rclpy/blob/4155fc0a201e94ce4317bb951b9c2b6e574c0881/rclpy/rclpy/__init__.py#L44) but you might have a reason for doing it beforehand. I can switch vendors without this by doing 
`RCLPY_IMPLEMENTATION=rmw_connext_cpp python3 src/rclpy/rclpy/test/rostopic_list.py` 
from the command line, for example